src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +91,13 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex() == Person.Sex.WOMAN
+                        && p.getAge() >= femaleAge)
+                .map(f -> f.getCats())
+                .flatMap(cl -> cl.stream())
+                .map(cat -> cat.getName())
+                .collect(Collectors.toList());
```

Comment:
you can write just `.toList();`
```suggestion
                .toList();
```